src/component/jhi-boolean-badge.component.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ Copyright 2013-2017 the original author or authors from the JHipster project.
+
+ This file is part of the JHipster project, see https://jhipster.github.io/
+ for more information.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+import { Component, Input, OnInit } from '@angular/core';
+import { JhiConfigService } from '../config.service';
+@Component({
+    selector: 'jhi-boolean-badge',
+    template: `<span class="badge" [ngClass]="clazz" [jhiTranslate]="translationKey"></span>`
+})
+export class JhiBooleanBadgeComponent implements OnInit {
+    @Input() value: boolean;
+    @Input() classBadgeTrue: string;
+    @Input() classBadgeFalse: string;
+    @Input() translationKeyTrue: string;
+    @Input() translationKeyFalse: string;
+    clazz: string;
+    translationKey: string;
+
+    classBadgeTrueConfig: string;
+    classBadgeFalseConfig: string;
+
+    constructor(configService: JhiConfigService) {
+        const config = configService.getConfig();
+        this.classBadgeTrueConfig = config.classBadgeTrue;
```

Comment:
can you check what happens if you assign `this.classBadgeTrue = config.classBadgeTrue` here and then override it in the component usage? I dont have the time to check but I guess the input assignment happens after constructor and if so should work without using `classBadgeTrueConfig` but I could be wrong